### ISSUE TITLE
spread uses "arguments" too agressively

### ISSUE THREAD

**Issue by loganfsmyth** [babel/babel#1796]
```
function fn2(){
  console.log(arguments.length);
}

function fn(...args){
  args.pop()
  fn2(...args);
}
```

becomes

```
function fn2() {
  console.log(arguments.length);
}

function fn() {
  for (var _len = arguments.length, args = Array(_len), _key = 0; _key < _len; _key++) {
    args[_key] = arguments[_key];
  }

  args.pop();
  fn2.apply(undefined, arguments);
}
```

In this case, `fn2.apply(undefined, arguments);` is incorrect and should be `fn2.apply(undefined, args);` since `args` was mutated.

Reported "fundon" on Slack.

**Comment by benjamingr** [comment]
I assume https://github.com/babel/babel/commit/b57a80ecae305cde415da13e6cb465e6c7438a01 ?

**Comment by loganfsmyth** [comment]
I assume, don't have time to explore at the moment but wanted to make sure it was being tracked.

**Comment by sebmck** [comment]
5.6.1 regression

**Comment by sebmck** [comment]
Fixed as of 5.6.2, thanks!
